### PR TITLE
#65531: Improvements on variable archive

### PR DESCRIPTION
**Pull request by MahieddineAk**
This PR includes the following improvements to the variable archive page (some changes need to be made in Builder):
- Clean up the old variable archive page that contains hardcoded Builder code and replace it with a properly created page using Builder.
- Fix logical issues with the flex-drawer on different devices that caused variable details to open improperly on different columns count.
- Add a hash to variable items to make them linkable. When clicked, a smooth transition goes to the variable item and opens its details.
- Add an ACF repeater to introduce sector terms descriptions (EN, FR).
- Automatically update sector terms from the ACF repeater upon post save.
- Provide a migration script that initializes the ACF repeater value (with empty descriptions) on all variable items that have sector terms selected.
- Make the variable archive page fully responsive.
- Translate UI elements.

Note:
The improvements to the filter panel is WIP and will be done in a separate task.

### CODE REVIEW

File: fw-child/template/query/variable-item.php [hunk 1]
```diff
@@ -59,35 +67,35 @@
...
-					
-					<a href="<?php echo home_url ( $dl_slug ); ?>?var_id=<?php echo $item['id']; ?>" class="btn btn-light rounded-pill px-4"><?php _e ( 'Download Data', 'cdc' ); ?></a>
-				</div>
+
+			<div class="col col-lg-8 offset-lg-0 col-14 offset-1 variable-details__links">
+				<?php
+				$map_slug = 'map/';
+				$dl_slug  = 'download/';
+
+				if ( $GLOBALS['fw']['current_lang_code'] == 'fr' ) {
```
Comment (xfrenette-crim) [inline fw-child/template/query/variable-item.php:81]: To be consistent with the rest of this file, we should use the triple `===`, and put `'fr'` on the left:
(I don't know why GitHub shows differences in indentation though, so ignore it...)

 ```suggestion
				if ( 'fr' === $GLOBALS['fw']['current_lang_code'] ) {
```

**Review by xfrenette-crim — CHANGES_REQUESTED** [review]
Most comments are for minors changes.

But one comment concerns the code specific to the "Variables" page that was inserted in the  `flex-drawer.js` file. This file contains a plugin used in different places. It should stay generic and not contain page specific code. A suggestion was given to fix this problem.

**Comment by MahieddineAk** [comment]
@xfrenette-crim all your feedback have been addressed, thanks for checking.

File: fw-child/resources/js/child-functions.js [hunk 2]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
     }
-
-    if ($('body').hasClass('post-type-archive-variable')) {
-      $(document).on('fw_query_success', function (e, query_item) {
-        let query_items = query_item.find('.fw-query-items');
-
-        if (query_items.data('flex_drawer') == undefined) {
-          query_items.flex_drawer({
+    
+    if ($( '.variable-archive-page' ).length > 0) {
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:249]: Maybe add a one line comment above this `if` to easily see to what it applies and what it globally does.

File: fw-child/resources/js/child-functions.js [hunk 3]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
         }
-      });
+
+        const media_queries = {
+          large_screen: '(min-width: 768px)',
+          medium_screen: '(min-width: 576px) and (max-width: 767px)',
+          small_screen: '(max-width: 575px)',
+        };
+
+        const update_screen_condition = function ( plugin_obj, screen ) {
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:269]: This function should have a description (with what are the arguments).

File: fw-child/resources/js/child-functions.js [hunk 4]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
+        let variable_item = drawer_item.find( '.variable-item' );
+
+        if (variable_item.length > 0) {
+          let variable_item_hash = variable_item.attr( 'id' );
+
+          if (variable_item_hash) { // Check if variable_item_hash is not empty
+            window.location.hash = variable_item_hash;
+          }
+        }
+      } );
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:339]: Should this whole block (`$( document ).on( 'fw_fd_open', ...`) be inside the `if ($( '.variable-archive-page' ).length > 0)`, else it seems to also run on every page that uses the drawer with the query plugin (like news or learning zone) ?

File: fw-child/resources/js/child-functions.js [hunk 5]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
+                  .find( '.flex-drawer-trigger' )
+                  .trigger( 'click' );
+              } );
+          }
+        }
+      } );
+
+      $( document ).on( 'fw_fd_open', function ( e, drawer_item ) {
+        // Update the URL hash if the variable item hash exists.
+        let variable_item = drawer_item.find( '.variable-item' );
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:330]:
```suggestion
        const variable_item = drawer_item.find( '.variable-item' );
```

File: fw-child/resources/js/child-functions.js [hunk 6]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
+
+        if (variable_item.length > 0) {
+          let variable_item_hash = variable_item.attr( 'id' );
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:333]:
```suggestion
          const variable_item_hash = variable_item.attr( 'id' );
```

File: fw-child/resources/js/child-functions.js [hunk 7]
```diff
@@ -245,20 +245,98 @@ const $ = jQuery;
...
-    if ($('body').hasClass('post-type-archive-variable')) {
-      $(document).on('fw_query_success', function (e, query_item) {
-        let query_items = query_item.find('.fw-query-items');
-
-        if (query_items.data('flex_drawer') == undefined) {
-          query_items.flex_drawer({
+    
+    if ($( '.variable-archive-page' ).length > 0) {
+      $( document ).on( 'fw_query_success', function ( e, query_item ) {
+        let query_items = query_item.find( '.fw-query-items' );
```
Comment (xfrenette-crim) [inline fw-child/resources/js/child-functions.js:251]:
```suggestion
        const query_items = query_item.find( '.fw-query-items' );
```

File: fw-child/resources/js/child-functions.js (hunk 4 above)
Comment (MahieddineAk) [reply fw-child/resources/js/child-functions.js:339]: @xfrenette-crim it is already inside it.

**Comment by MahieddineAk** [comment]
@xfrenette-crim all your new feedback has been addressed.

File: fw-child/resources/js/child-functions.js (hunk 4 above)
Comment (xfrenette-crim) [reply fw-child/resources/js/child-functions.js:339]: You're right. My mistake.

**Review by xfrenette-crim — APPROVED** [review]

**Pull request merged**